scripts/simex

```diff
@@ -243,41 +278,64 @@ builds_subcmds.required = True
+				[build.info for build in sel[revision]], [], [])
 
-builds_make_parser = builds_subcmds.add_parser('make')
+builds_make_parser = builds_subcmds.add_parser('make', parents=[build_selection_parser])
 builds_make_parser.set_defaults(cmd=do_builds_make)
 
+def do_builds_purge(args):
+	cfg = extl.base.config_for_dir()
+
+	purge_builds_from(args, cfg.all_non_dev_builds())
+
+builds_purge_parser = builds_subcmds.add_parser('purge', parents=[build_selection_parser])
+builds_purge_parser.set_defaults(cmd=do_builds_purge)
+builds_purge_parser.add_argument('-f', action='store_true', help='Execute purge')
+
+def do_builds_remake(args):
+	cfg = extl.base.config_for_dir()
+	args.recheckout = True
+	args.f = True
+
+	wanted_phases = select_phases_from_cli(args)
+	ordered_revisions, sel = select_builds_from_cli(args, cfg.all_non_dev_builds())
+	for revision in ordered_revisions:
+
+		simexpal.build.make_builds(cfg, revision,
+				[build.info for build in sel[revision]], [build.name for build in sel[revision]], wanted_phases)
+
+builds_remake_parser = builds_subcmds.add_parser('remake', parents=[build_selection_parser])
+builds_remake_parser.set_defaults(cmd=do_builds_remake)
+
 # ---------------------------------------------------------------------------------------
 
 def do_develop(args):
 	cfg = extl.base.config_for_dir()
 
-	if args.revision is not None:
-		revision = cfg.get_revision(args.revision)
-	else:
-		revision = cfg.get_revision(simexpal.base.DEFAULT_DEV_BUILD_NAME)
-
-	if not revision.is_dev_build:
-		print("Revision '{}' is not a dev-build.".format(revision.name), file=sys.stderr)
+	wanted_phases = select_phases_from_cli(args)
+	if not wanted_phases:
 		return
-	else:
-		wanted_phases = select_phases_from_cli(args)
 
-		if not wanted_phases:
-			return
+	if args.purge:
+		purge_builds_from(args, cfg.all_dev_builds(), args.delete_source)
 
-		simexpal.build.make_builds(cfg, revision,
-				[cfg.get_build(build, revision).info for build in args.builds], args.builds, wanted_phases)
+	else:
+		ordered_revisions, sel = select_builds_from_cli(args, cfg.all_dev_builds())
+		for revision in ordered_revisions:
+
+			simexpal.build.make_builds(cfg, revision,
+					[build.info for build in sel[revision]], [build.name for build in sel[revision]], wanted_phases)
 
 dev_builds_parser = main_subcmds.add_parser('develop', help='Build local programs',
-		aliases=['d'], parents=[phase_selection_parser])
+		aliases=['d'], parents=[phase_selection_parser, build_selection_parser])
 dev_builds_parser.set_defaults(cmd=do_develop)
-dev_builds_parser.add_argument('--revision', type=str)
-dev_builds_parser.add_argument('builds', nargs='+', type=str)
+dev_builds_parser.add_argument('--purge', action='store_true', help='Purge selected dev-builds')
+dev_builds_parser.add_argument('-f', action='store_true', help='Confirm action')
+dev_builds_parser.add_argument('--delete_source', action='store_true', help='Deletes the source directory when purging')
```

Comment:
Parameters should use hyphens, not underscores.

Reply:
Done